tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/ShapeOps.java

```diff
@@ -0,0 +1,329 @@
+ */
+@Operator
+public abstract class ShapeOps {
+
+    /**
+     * flatten the shape to 1 dimension
+     *
+     * @param <T> the type of operand
+     * @param scope current scope
+     * @param operand the operand to flatten
+     * @return the reshaped operand
+     */
+    @Endpoint(name = "flatten")
+    public static <T extends TType> Operand<T> flatten(Scope scope, Operand<T> operand) {
+        return flatten(scope, operand, TInt32.DTYPE);
+    }
+
+    /**
+     * flatten the shape to 1 dimension
+     *
+     * @param <T> the type of operand
+     * @param <U> the shape datatype.
+     * @param scope current scope
+     * @param operand the operand to flatten
+     * @param dType the shape datatype.
+     * @return the reshaped operand
+     */
+    @Endpoint(name = "flatten")
+    public static <T extends TType, U extends TNumber> Operand<T> flatten(Scope scope, Operand<T> operand, DataType<U> dType) {
+        Operand<U> flatShape = flatten(scope, Shape.create(scope, operand, dType), dType);
+        return Reshape.create(scope, operand, flatShape);
+    }
+
+    /**
+     * flatten the shape to 1 dimension
+     *
+     * @param scope current scope
+     * @param shape the TensorFlow shape
+     * @return the flattened shape
+     * @see reduceDims
+     */
+    @Endpoint(name = "flatten")
+    public static Operand<TInt32> flatten(Scope scope, Shape<TInt32> shape) {
+        return flatten(scope, shape, TInt32.DTYPE);
+    }
+
+    /**
+     * flatten the shape to 1 dimension
+     *
+     * @param <U> the shape datatype.
+     * @param scope current scope
+     * @param shape the TensorFlow shape
+     * @param dType the shape datatype.
+     * @return the flattened shape
+     * @see reduceDims
+     */
+    @Endpoint(name = "flatten")
+    public static <U extends TNumber> Operand<U> flatten(Scope scope, Shape<U> shape, DataType<U> dType) {
+        return ExpandDims.create(scope,
+                size(scope, shape, dType),
+                Cast.create(scope, Constant.scalarOf(scope, -1), TInt32.DTYPE));
+    }
+
+    /**
+     * get the size represented by the TensorFlow shape
+     *
+     * @param scope current scope
+     * @param shape the TensorFlow shape
+     * @return the size
+     */
+    @Endpoint(name = "size")
+    public static Operand<TInt32> size(Scope scope, Shape<TInt32> shape) {
+        return size(scope, shape, TInt32.DTYPE);
+    }
+
+    /**
+     * get the size represented by the TensorFlow shape
+     *
+     * @param <U> the shape datatype. the type of the shape
+     * @param scope current scope
+     * @param shape the TensorFlow shape
+     * @param dType the shape datatype.
+     * @return the size
+     */
+    @Endpoint(name = "size")
+    public static <U extends TNumber> Operand<U> size(Scope scope, Shape<U> shape, DataType<U> dType) {
+        Slice<U> dims = Slice.create(scope, shape,
+                Cast.create(scope, Constant.arrayOf(scope, (new int[]{0})), dType),
```

Comment:
Parentheses not required around the array instantiation

Reply:
OK